cpp/models/abm/person.cpp

```diff
@@ -44,75 +44,58 @@ Person::Person(LocationId id, InfectionProperties infection_properties, AgeGroup
     m_random_goto_work_hour   = UniformDistribution<double>::get_instance()();
     m_random_goto_school_hour = UniformDistribution<double>::get_instance()();
 
-    if (infection_properties.state == InfectionState::Infected && infection_properties.detected) {
+    if (infection != nullptr && infection->is_detected()) {
         m_quarantine = true;
     }
-    if (infection_properties.state == InfectionState::Exposed) {
-        m_time_until_carrier = hours(UniformIntDistribution<int>::get_instance()(
-            0, int(global_params.get<IncubationPeriod>()[{m_age, m_vaccination_state}] * 24)));
-    }
 }
 
-Person::Person(Location& location, InfectionProperties infection_properties, AgeGroup age,
-               const GlobalInfectionParameters& global_params, VaccinationState vaccination_state, uint32_t person_id)
-    : Person({location.get_index(), location.get_type()}, infection_properties, age, global_params, vaccination_state,
-             person_id)
+Person::Person(const Location& location, const AgeGroup& age, const VaccinationState& vaccination_state,
+               Infection* const infection, const uint32_t person_id)
+    : Person({location.get_index(), location.get_type()}, age, vaccination_state,
+             infection, person_id)
 {
 }
 
-void Person::interact(TimeSpan dt, const GlobalInfectionParameters& global_infection_params, Location& loc,
-                      const GlobalTestingParameters& global_testing_params)
+void Person::interact(const TimePoint& t, const TimeSpan& dt,
+              const GlobalInfectionParameters& global_infection_parameters, Location& loc)
```

Comment:
Do we actually still need the `global_infection_parameters` here?

Reply:
Maybe, maybe not. I have already commented it out in the `Location::interact`, but this is not finished. In my view, global infection parameters should be used in the drawing at the start of each infection, where it would be necessary to pass them all the way to where a new infection is created, which would be exactly in `Location::interact` or a function that is called from there. We can discuss this.

Reply:
Update on this: Yes, we do need them to draw the infection.